### DISCUSSION TITLE
Cache.append

### DISCUSSION THREAD

**Discussion by Bondian** [Chimildic/goofy#139] [Принцип работы]
При использовании Cache.append массив для записи данных в файл сам пополняется данными из этого файла. Это ошибка или свойство этой функции?
Для наглядности, что я имею ввиду:
```
  let tracks = Source.getTracks([
    { name: 'a-b-c tracks', id: '...' },
  ]);
  Cache.append('x-y-z tracks.json', tracks, 'begin');

  Playlist.saveWithReplace({
    name: 'a-b-c-x-y-z',
    public: false,
    tracks: tracks,
    // description: 'в массив добавились треки из файла',
  });
```

**Comment by Chimildic** [comment]
> Это ошибка или свойство этой функции?

Намеренное допущение. Конечно хорошим тоном будет не изменять входные данные у любых функций. Но порой это полезно. Считайте неявный возврат всех данных с файла включая новые.

Чем это мешает в вашем случае?

**Reply by Bondian** [reply]
Составлял шаблон, в котором исходные данные изменялись и, по мере изменения, поочерёдно добавлялись в разные файлы. Но он не работал по задуманное логике. Пришлось разбираться. А оно "вононочто" 🙂
